src/Zynga/Framework/Factory/V2/Base.hh

```diff
@@ -168,7 +175,18 @@ abstract class Base implements FactoryInterface {
 
   public static function disableMockDrivers(): bool {
     $template = self::getFactoryTemplate();
+    self::clearOverridenMockDrivers();
     return $template->disableMockDrivers();
   }
 
+  public static function overrideMockDriver(
+    string $actualClass,
+    string $mockClass,
+  ): void {
+    self::$_mockOverrides[$actualClass] = $mockClass;
```

Comment:
I do not think we need to check if the key exists, since if we were to change the override for the same key twice. Since we would always want to use the most recent override i think.